mobile/src/main/java/rs/readahead/washington/mobile/views/activity/viewer/PhotoViewerActivity.kt

```diff
@@ -0,0 +1,320 @@
+package rs.readahead.washington.mobile.views.activity.viewer
+
+import android.content.Intent
+import android.graphics.drawable.Drawable
+import android.os.Build
+import android.os.Bundle
+import android.view.View
+import androidx.activity.viewModels
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.widget.Toolbar
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.DataSource
+import com.bumptech.glide.load.engine.DiskCacheStrategy
+import com.bumptech.glide.load.engine.GlideException
+import com.bumptech.glide.request.RequestListener
+import com.bumptech.glide.request.target.Target
+import com.google.android.exoplayer2.ui.StyledPlayerView
+import com.hzontal.tella_vault.Metadata
+import com.hzontal.tella_vault.VaultFile
+import org.hzontal.shared_ui.bottomsheet.BottomSheetUtils
+import org.hzontal.shared_ui.utils.DialogUtils
+import rs.readahead.washington.mobile.MyApplication
+import rs.readahead.washington.mobile.R
+import rs.readahead.washington.mobile.bus.event.MediaFileDeletedEvent
+import rs.readahead.washington.mobile.bus.event.VaultFileRenameEvent
+import rs.readahead.washington.mobile.databinding.ActivityPhotoViewerBinding
+import rs.readahead.washington.mobile.media.MediaFileHandler
+import rs.readahead.washington.mobile.views.activity.MetadataViewerActivity
+import rs.readahead.washington.mobile.views.activity.viewer.VaultActionsHelper.initContracts
+import rs.readahead.washington.mobile.views.activity.viewer.VaultActionsHelper.showVaultActionsDialog
+import rs.readahead.washington.mobile.views.base_ui.BaseLockActivity
+
+
+class PhotoViewerActivity : BaseLockActivity(), StyledPlayerView.ControllerVisibilityListener {
+    companion object {
+        const val VIEW_PHOTO = "vp"
+        const val NO_ACTIONS = "na"
+    }
+
+    private val viewModel: SharedMediaFileViewModel by viewModels()
+    private lateinit var toolbar: Toolbar
+    private var vaultFile: VaultFile? = null
+    private var showActions = false
+    private var actionsDisabled = false
+    private var alertDialog: AlertDialog? = null
+    private var isInfoShown = false
+    private lateinit var binding: ActivityPhotoViewerBinding
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityPhotoViewerBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        initContracts()
+        setupToolbar()
+        initObservers()
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            binding.toolbar.outlineProvider = null
```

Comment:
There should be an extension function in Djordje's PR now, that you can use configure the appBar